sage-ios/SAGE/SAGE/EditAnnouncementController.swift

```diff
@@ -52,4 +58,22 @@ class EditAnnouncementController: AddAnnouncementController {
         }
     }
 
+    func deleteAnnouncement(sender: UIButton!) {
+        let view = (self.view as! AddAnnouncementView)
```

Comment:
you already have the addAnnouncement view property you can just use that directly